python_modules/libraries/dagster-components/dagster_components/core/schema/context.py

```diff
@@ -42,10 +42,13 @@ def with_scope(self, **additional_scope) -> "ResolutionContext":
 
     def _resolve_inner_value(self, val: Any) -> Any:
         """Resolves a single value, if it is a templated string."""
-        return NativeTemplate(val).render(**self.scope) if isinstance(val, str) else val
-
-    @overload
-    def resolve_value(self, val: ResolvableModel[T]) -> T: ...
+        if isinstance(val, ResolvableModel):
+            resolver = val.__dagster_resolver__(val)
```

Comment:
Minor thing: Can you make this `resolver_for_model` or something to consolidate all references to that class var in one spot so it is easy to change